src/common/components/repositories-home/index.js

```diff
@@ -46,12 +56,29 @@ class RepositoriesHome extends Component {
       </div>
     );
   }
+
+  renderSpinner() {
+    return (
+      <div className={ spinnerStyles }>
+        <Spinner />
+      </div>
+    );
+  }
+
+  render() {
+    // show spinner until we know if user has any enabled repos
+    return this.props.snaps.success
```

Comment:
I decided to show spinner until we know if user has any enabled repos, so the empty UI of repositories list doesn't pop up for a second.